src/map.rs

```diff
@@ -89,11 +88,11 @@ impl Map {
 }
 
 impl Map {
-    pub(crate) fn get_tile_by_gid(&self, gid: Gid) -> Option<(&Tile, &Tileset)> {
+    pub(crate) fn get_tileset_for_gid(&self, gid: Gid) -> Option<&Tileset> {
         self.tilesets
             .iter()
-            .filter_map(|ts| ts.get_tile_by_gid(gid).map(|tile| (tile, ts)))
-            .nth(0)
+            .rev()
+            .find(|ts| ts.first_gid <= gid.0)
```

Comment:
This assumes that tilesets are ordered by first GID. Should be documented.

Reply:
This is documented at https://doc.mapeditor.org/en/stable/reference/tmx-map-format/#tmx-tileset:
> If there are multiple <tileset> elements, they are in ascending order of their firstgid attribute.